de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/effortissue/EffortIssueRepository.java

```diff
@@ -0,0 +1,135 @@
+{
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final ExternalIssueDetailsRepository externalIssueDetailsRepository;
+
+	public EffortIssueRepository(final ExternalIssueDetailsRepository externalIssueDetailsRepository)
+	{
+		this.externalIssueDetailsRepository = externalIssueDetailsRepository;
+	}
+
+	public EffortIssue store(@NonNull final EffortIssue effortIssue)
+	{
+		final I_S_Effort_Issue record = effortIssue.getEffortIssueId() != null
+				? InterfaceWrapperHelper.load(effortIssue.getEffortIssueId(), I_S_Effort_Issue.class)
+				: InterfaceWrapperHelper.newInstance(I_S_Effort_Issue.class);
+
+		record.setAD_Org_ID(effortIssue.getOrgId().getRepoId());
+
+		record.setName(effortIssue.getName());
+		record.setValue(effortIssue.getName());
+		record.setDescription(effortIssue.getDescription());
+
+		record.setAD_User_ID(NumberUtils.asInt(effortIssue.getAssigneeId(), -1));
+		record.setS_Milestone_ID(NumberUtils.asInt(effortIssue.getMilestoneId(), -1));
+		record.setEstimatedEffort(effortIssue.getEstimatedEffort());
+		record.setBudgetedEffort(effortIssue.getBudgetedEffort());
+		record.setEffort_UOM_ID(effortIssue.getEffortUomId().getRepoId());
+
+		record.setProcessed(effortIssue.isProcessed());
+
+		record.setExternalId(effortIssue.getExternalIssueId());
+		record.setIssueURL(effortIssue.getExternalIssueURL());
+		record.setExternalIssueNo(effortIssue.getExternalIssueNo());
+
+		InterfaceWrapperHelper.save(record);
+
+		effortIssue.setEffortIssueId(EffortIssueId.ofRepoId(record.getS_Effort_Issue_ID()));
+
+		persistIssueDetails(effortIssue);
+
+		return effortIssue;
+	}
+
+	public Optional<EffortIssue> getEntityByExternalId(@NonNull final String externalIssueId)
+	{
+
+		final I_S_Effort_Issue effort_issue = queryBL
+				.createQueryBuilder(I_S_Effort_Issue.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_S_Effort_Issue.COLUMNNAME_ExternalId, externalIssueId)
+				.create()
+				.firstOnly(I_S_Effort_Issue.class);
+
+		return effort_issue != null ? Optional.of(buildEffortIssueEntity(effort_issue)) : Optional.empty();
+	}
+
+
+	private EffortIssue buildEffortIssueEntity(final I_S_Effort_Issue record)
+	{
+		return EffortIssue.builder()
+				.effortIssueId(EffortIssueId.ofRepoId(record.getS_Effort_Issue_ID()))
+				.orgId(OrgId.ofRepoId(record.getAD_Org_ID()))
+				.milestoneId(MilestoneId.ofRepoIdOrNull(record.getS_Milestone_ID()))
+				.assigneeId(UserId.ofNullableRepoId(record.getAD_User_ID()))
+				.name(record.getName())
+				.description(record.getDescription())
+				.processed(record.isProcessed())
+				.estimatedEffort(record.getEstimatedEffort())
+				.budgetedEffort(record.getBudgetedEffort())
+				.effortUomId(UomId.ofRepoId(record.getEffort_UOM_ID()))
+				.externalIssueId(record.getExternalId())
+				.externalIssueNo(record.getExternalIssueNo())
+				.externalIssueURL(record.getIssueURL())
+				.build();
+	}
+
+	private void persistIssueDetails(final EffortIssue effortIssue)
```

Comment:
instead of passing the whole EffortIssue, why not pass the EffortIssueId and the details?

Reply:
no reason, I'll go with your suggestion